x-pack/plugins/synthetics/server/routes/monitor_cruds/add_monitor_project_legacy.ts

```diff
@@ -41,7 +41,8 @@ export const addSyntheticsProjectMonitorRouteLegacy: SyntheticsStreamingRouteFac
   }): Promise<any> => {
     try {
       const monitors = (request.body?.monitors as ProjectMonitor[]) || [];
-      const spaceId = server.spaces.spacesService.getSpaceId(request);
+      const { id: spaceId } = await server.spaces.spacesService.getActiveSpace(request);
```

Comment:
I think we should do something along the lines of 
```
if (!spaceId) {
        return response.notFound({ body: { message: 'Kibana space does not exist' } });
}
```
Or something along that lines.

For add_monitor_project and add_monitor_project_legacy

Reply:
I couldn't do this for the legacy API since those requests are always 200 (I don't have enough context to know why), but after checking in the slack channel what I did there was to report all monitors as failed and add a message with a reason why.

Reply:
my view is that, we don't bother doing it in legacy rout and leave it as is.

Reply:
or better we just remove the legacy route, that will force users to upgrade for better.